src/Services/PushService.ts

```diff
@@ -0,0 +1,80 @@
+import firebase from 'react-native-firebase';
+import {Alert} from 'react-native';
+import AsyncStorage from '@react-native-community/async-storage';
+
+let notificationListener;
+let notificationOpenedListener;
+let messageListener;
+
+export async function createNotificationListeners() {
+    /*
+     * Triggered when a particular notification has been received in foreground
+     * */
+    notificationListener = firebase.notifications().onNotification((notification) => {
+        const {title, body} = notification;
+        showAlert(title, body);
+    });
+
+    /*
+     * If your app is in background, you can listen for when a notification is clicked / tapped / opened as follows:
+     * */
+    notificationOpenedListener = firebase
+        .notifications()
+        .onNotificationOpened((notificationOpen) => {
+            const {title, body} = notificationOpen.notification;
+            showAlert(title, body);
+        });
+
+    /*
+     * If your app is closed, you can check if it was opened by a notification being clicked / tapped / opened as follows:
+     * */
+    const notificationOpen = await firebase.notifications().getInitialNotification();
+    if (notificationOpen) {
+        const {title, body} = notificationOpen.notification;
+        showAlert(title, body);
+    }
+    /*
+     * Triggered for data only payload in foreground
+     * */
+    messageListener = firebase.messaging().onMessage((message) => {
+        //process data message
+        console.log(JSON.stringify(message));
+    });
+}
+
+function showAlert(title: string, body: string) {
+    Alert.alert(title, body, [{text: 'OK', onPress: () => console.log('OK Pressed')}], {
+        cancelable: false,
+    });
+}
+
+export async function checkPermission() {
+    const enabled = await firebase.messaging().hasPermission();
+    if (enabled) {
+        getToken();
+    } else {
+        requestPermission();
+    }
+}
```

Comment:
I like this PushService file, but I would make this a method called hasPermission and return just whether firebase has the permissions. Then the logic can go in the componentDidMount. If it doesn't have the permission, request it and after that we get and store the token.